include/aws/common/rw_lock.h

```diff
@@ -54,6 +54,7 @@ AWS_COMMON_API int aws_rw_lock_wlock(struct aws_rw_lock *lock);
  * Attempts to acquire the lock but returns immediately if it can not.
  * While on some platforms such as Windows, this may behave as a reentrant mutex,
  * you should not treat it like one. On platforms it is possible for it to be non-reentrant, it will be.
+ * Note: For windows, minimum support server version is Windows Server 2008 R2 [desktop apps | UWP apps]
```

Comment:
Just out of curiosity, why not fallback to ```Enter/LeaveCriticalSection``` which have been around since the 90s?

Reply:
hmmm, good call! So, if `TryAcquireSRWLockExclusive` not exist, we could use `TryEnterCriticalSection` for try lock and `Enter/LeaveCriticalSection` for lock and unlock right?

Reply:
Use critical section will:
- Will increase the size of lock for modern windows (We don't take allocator for mutex)
- We have static allocator for mutex…